src/sentry/workflow_engine/migration_helpers/rule_action.py

```diff
@@ -62,75 +19,88 @@ def build_notification_actions_from_rule_data_actions(
 
     :param actions: list of action data (Rule.data.actions)
     :return: list of notification actions (Action)
+
+    :raises ValueError: if action is missing an id
+    :raises KeyError: if there isn't a translator registered for the id
     """
 
     notification_actions: list[Action] = []
 
     for action in actions:
-        # Use Rule.integration.provider to get the action type
-        action_type = RULE_REGISTRY_ID_TO_INTEGRATION_PROVIDER.get(action.get("id"))
-        if action_type is None:
-            _logger.warning(
-                "Action type not found for action",
+        registry_id = action.get("id")
+        if not registry_id:
+            logger.error(
+                "No registry ID found for action",
+                extra={"action_uuid": action.get("uuid")},
+            )
+            continue
+
+        try:
+            translator_class = issue_alert_action_translator_registry.get(registry_id)
+            translator = translator_class()
```

Comment:
maybe you can pass in the action to the translator class so you can do this in one go

```
target_display = None
if translator.target_display_key:
    target_display = action.get(translator.target_display_key)
```
becomes
```
target_display = translator.target_display_key
```